src/content/docs/en/reference/modules/astro-actions.mdx

```diff
@@ -301,3 +301,46 @@ export const onRequest = defineMiddleware(async (context, next) => {
 </p>
 
 `deserializeActionResult()` will reverse the effect of `serializeActionResult()` and return an action result to its original state. This is useful to access the `data` and `error` objects on a serialized action result.
+
+### `getActionPath()`
+
+<p>
+
+**Type:** `(action: ActionClient<any, any, any>) => string`
+<Since v="5.1.0" />
+</p>
+
+The `getActionPath()` utility accepts an action and returns a URL path so you can execute an action call as a `fetch()` operation directly. This allows you to provide details such as custom headers when you call your action.
+
+This example shows how to call a defined `like` action passing the `Authorization` header and the [`keepalive`](https://developer.mozilla.org/en-US/docs/Web/API/Request/keepalive) option:
+
+```astro title="src/components/my-component.astro" {8,11}
+<script>
+import { actions, getActionPath } from 'astro:actions'
+
+await fetch(getActionPath(actions.like), {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json',
+    Authorization: 'Bearer YOUR_TOKEN'
+  },
+  body: JSON.stringify({ id: 'YOUR_ID' }),
+  keepalive: true
+})
+</script>
+```
+
+This example shows how to call the same `like` action using the [`sendBeacon`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon) API:
+
+```astro title="src/components/my-component.astro" {5} "sendBeacon"
+<script>
+import { actions, getActionPath } from 'astro:actions'
+
+navigator.sendBeacon(
+  getActionPath(actions.like),
+  new Blob([JSON.stringify({ id: 'YOUR_ID' })], {
+    type: 'application/json'
+  })
+)
+</script>
+```
```

Comment:
@sarah11918 see https://github.com/withastro/astro/pull/12721/files#r1888930180
``````suggestion
```

:::note
Actions do not return plain JSON. Use `deserializeActionResult` from `astro:actions` if you need the returned data
:::

``````

Reply:
This will work in a pinch, but we don't actually have any code showing `deserializeActionResult` in use (not even the API reference). This could be a good place for that!

If you or @bholmesdev can whip something up here, then I think showing it in use is *better* than having a note that says "use it" with a link to something that similarly doesn't show how to use it.

Reply:
I've not linked it on purpose, because the documented `deserializeActionResult` is the one returned from `getActionContext` above in the reference. This one we refer to is directly exported from `astro:actions`

Reply:
I think what we changed in the changeset works, actually! Let me suggest that here!